### ISSUE TITLE
Internal or external parsed entities in the external DTD subset of an SVG content document

### ISSUE THREAD

**Issue by murata2makoto** [w3c/epub-specs#1357]
Consider this XML document.

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE svg SYSTEM "mySvg.dtd">
<svg xmlns="http://www.w3.org/2000/svg" width="100%" height="100%">
    &desc;
    <g alignment-baseline="baseline"></g>
</svg>
```

If mySvg.dtd contains `<!ENTITY desc "<desc></desc>">`, then `desc` is an internal parsed entity defined in an external DTD subset.  If mySvg.dtd contains `<!ENTITY desc SYSTEM "desc.ent">`, then `desc` is an external parsed entity defined in an external DTD subset.  

This document has not been allowed as an SVG content document.  T[he currently proposed text](https://github.com/w3c/epub-specs/issues/1338#issuecomment-715919230) allows it, however.  Moreover, an EPUB RS might or might not resolve the entity reference. 

I don't think that it is a good idea to allow this as an SVG content document.

**Comment by iherman** [comment]
Note that, I believe this file would not be accepted by web browsers. I tested the other examples, and they were all considered valid and displayed correctly (I have modified them slightly to produce a visible result) this example is not accepted due to a missing entity.

**Comment by murata2makoto** [comment]
@iherman 

I suppose that you created mySvg.dtd.  Even if you do, those non-validating processors which do not fetch external DTD subsets will fail to handle this document.  Those which do will not fail.  This lack of interoperability is a shame of XML.

**Comment by iherman** [comment]
> I suppose that you created mySvg.dtd. 

Yes, I did.

> Even if you do, those non-validating processors which do not fetch external DTD subsets will fail to handle this document. Those which do will not fail. This lack of interoperability is a shame of XML.

:-(

However: if web browsers do not do this, this means that no Reading System, unless they develop and keep up-to-date their own rendering engine, will handle that either.

**Comment by murata2makoto** [comment]
EPUB 3 has disallowed this problematic example.  I would like to continue to disallow it.

**Comment by iherman** [comment]
This issue was discussed in [a meeting](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2020-11-06-epub#section3).

- `RESOLVED:  Merge PR #1368 to address outstanding DTD issues, and close GH issues 1369-1373`
<details><summary><i class="differentiate">View the transcript</i></summary>
Wendy Reid: we had resolutions at the F2F, and further discussions on github<br/>
… and came to a happy place<br/>
Matt Garrish: <a href="https://github.com/w3c/epub-specs/pull/1368">https://github.com/w3c/epub-specs/pull/1368</a><br/>
Matt Garrish: where we ended up was…<br/>
… we put in an allowance for a specific set of external identifiers that we have put in an appendix<br/>
… we have SVG and MathML that are allowed to be used in content docs or in separate files<br/>
… and we made a restriction against external entities in the internal DTD subset<br/>
… so it prevents some security issues but eases authoring<br/>
… so we’ll no longer force people to remove SVG DTDs from tool-generated files<br/>
… I’m hoping this is it :)<br/>
Ivan Herman: tech comment<br/>
… in fact, the changes are such that<br/>
… makes possible something that I’m not sure we really use<br/>
… I can define as part of an internal entity something that won’t go out to the network<br/>
… I’m not sure if this feature is in use<br/>
… formal comment<br/>
… there was a formal resolution on the previous version; this PR slightly changes that<br/>
… can we get a formal resolution to merge, and also close a bunch of issues which were examples of the problem?<br/>
Proposed resolution: Merge PR #1368 to address outstanding DTD issues, and close GH issues 1369-1373 (Wendy Reid)<br/>
Garth Conboy: +1<br/>
Matt Garrish: +1<br/>
Ivan Herman: +1<br/>
Charles LaPierre: +1<br/>
Matthew Chan: +1<br/>
Wendy Reid: +1<br/>
Brady Duga: +1<br/>
George Kerscher: +1<br/>
Laura Brady: +1<br/>
Bill Kasdorf: +1<br/>
Ben Schroeter: +1<br/>
Resolution #1: Merge PR #1368 to address outstanding DTD issues, and close GH issues 1369-1373<br/>
</details>

**Comment by iherman** [comment]
@wareid I believe this issue can also be closed
